src/msg/Messenger.h

```diff
@@ -419,11 +429,10 @@ class Messenger {
    */
   void add_dispatcher_tail(Dispatcher *d, PriorityDispatcher::priority_t priority=Dispatcher::PRIORITY_DEFAULT) {
     bool first = dispatchers.empty();
-    dispatchers.push_back(PriorityDispatcher{priority, d});
-    std::stable_sort(dispatchers.begin(), dispatchers.end());
+    const PriorityDispatcher entry{priority, d};
```

Comment:
has the valgrind issue spooked you into using `const` here to avoid a move? :laughing: